unirec/unirec.c

```diff
@@ -804,12 +819,19 @@ ur_template_t *ur_create_template(const char *fields, char **errstr)
 {
    // Count number of fields
    int n_fields = 0, written_fields = 0;
-   const char *tmp = fields;
-   if (fields != NULL && fields[0] != '\0') {
-      n_fields = 1;
-      while (*tmp != '\0') {
-         if (*(tmp++) == ',') {
-            n_fields++;
+   if (fields) {
+      /* skip leading spaces */
+      while (isspace(*fields) && *fields != '\0') {
```

Comment:
According to `isspace` docs, '\0' is not considered whitespace, so the whole `*fields != '\0'` is unnecessary. However, I still think it's good to keep that explicit check here. And if you think it should be first, than OK, I'll move it.